bash-5.1/execute_cmd.c

```diff
@@ -5592,14 +5594,17 @@ execute_disk_command (words, redirects, command_line, pipe_in, pipe_out,
 	  exit (execute_shell_function (hookf, wl));
 	}
 
+#if defined (BASH_SHELL_EXECVE_PLUGIN)
+      /*get original user input args for plugin*/
+      char **original_args = strvec_from_word_list (original_words, 0, 0, (int *)NULL);
+      result = invoke_plugin_on_shell_execve (current_user.user_name, command, original_args);
+      xfree(original_args);
+
       /* Execve expects the command name to be in args[0].  So we
 	 leave it there, in the same format that the user used to
 	 type it in. */
       args = strvec_from_word_list (words, 0, 0, (int *)NULL);
```

Comment:
>args [](http://example.com/codeflow?start=6&length=4)

Is it needed to `xfree(args)`?
If it is a upstream bug, let's create a separate patch and try upstream the bugfix.

Reply:
Seems not a bug, bash will exit after using args:

      exit (shell_execve (command, args, export_env));